### PR TITLE
DM-16650: update to new nJy PhotoCalib definition

### PR DESCRIPTION
**Pull request by parejkoj**

### CODE REVIEW

File: python/lsst/jointcal/utils.py [hunk 1]
```diff
@@ -350,23 +349,23 @@ def _make_match_dict(self, reference, visit_catalogs, photoCalibs, refcalib=None
...
-            maggiesToJansky = 3631
             flux = match[1]['slot_CalibFlux_instFlux']
             if flux < 0:
                 return None
             else:
-                flux = maggiesToJansky * photoCalib.instFluxToMaggies(match[1], "slot_CalibFlux").value
+                flux = photoCalib.instFluxToNanojansky(match[1], "slot_CalibFlux").value
 
-            # NOTE: Have to protect against negative reference fluxes too.
+            # NOTE: Have to protect against negative "reference" fluxes too.
```
Comment (erykoff) [inline python/lsst/jointcal/utils.py]: Off-topic, but can you use a sourceSelector on the reference catalog for a s/n or flux>0 cut?

File: python/lsst/jointcal/utils.py [hunk 2]
```diff
@@ -350,23 +349,23 @@ def _make_match_dict(self, reference, visit_catalogs, photoCalibs, refcalib=None
...
-                    ref_flux = maggiesToJansky * photoCalib.instFluxToMaggies(match[0], ref_flux_key).value
+                    ref_flux = refcalib.instFluxToNanojansky(match[0], ref_flux_key).value
             else:
-                # a.net fluxes are already in Janskys.
-                ref_flux = match[0][ref_flux_key.format(filt)]
+                # refcat fluxes are already in Janskys.
+                # TODO: Once RFC-549 is fully implemented, we can remove the 1e9 prefactor.
+                ref_flux = 1e9 * match[0][ref_flux_key.format(filt)]
                 if ref_flux < 0:
                     return None
```
Comment (erykoff) [inline python/lsst/jointcal/utils.py]: Also a bit off-topic, and I don't know how long jointcal spends doing this collation, but I think this whole section can be numpy-array-ized and would avoid a lot of logic and loops.  In the end the return value of `_make_match_dict` is numpy arrays anyway.

Comment (parejkoj) [reply python/lsst/jointcal/utils.py]: Yeah... I'm not at all happy about the code here in `utils.py`. I wrote it when I was trying to come up with a way to verify that joincal was doing better than processCcd, and it ended up a big mess over time. It is used in jointcal's test suite, but not elsewhere and the plots it can make have helped me debug things in the past.

However, you are correct, the calculations done in this file are a non-trivial part of the time running jointcal's tests due to the crappy way they were implemented. Ideally, I'd throw almost all of this away and just call the AM1, PA1 calculations from validate_drp. That is sort of DM-6577, but not exactly: I can't make validate_drp a dependency of jointcal, and the question of where to put the metric calculation code has repeatedly stalled.

There's talk of doing a hack week at Princeton to develop some better metrics and analysis of jointcal's output: we might be able to replace all of this then.

Long story short: I completely agree, but it's too much work for this ticket.
